### PR TITLE
MaximumLikelihoodFactory enhancements

### PR DESCRIPTION
**Pull request by jschueller**
- the constraint was not taken into account
- factories using MLF did not take their solver attribute into account, removed it in favor of an RM key to set the MLF solver

### CODE REVIEW

File: lib/src/Uncertainty/Distribution/TrapezoidalFactory.cxx [hunk 1]
```diff
@@ -131,7 +110,17 @@ Trapezoidal TrapezoidalFactory::buildAsTrapezoidal(const Sample & sample) const
...
+  input[0] = "a";
+  input[1] = "b";
+  input[2] = "c";
+  input[3] = "d";
+  Description formula(3);
+  formula[0] = "b - a";// a <= b
+  formula[1] = OSS() << "c - b - " << SpecFunc::ScalarEpsilon;// b < c
+  formula[2] = "d - c";// c <= d
+  SymbolicFunction constraint(input, formula);
+  factory.setOptimizationInequalityConstraint(constraint);
```
Comment (sofianehaddad) [inline lib/src/Uncertainty/Distribution/TrapezoidalFactory.cxx]: hmm what about if Muparser is missing ?

Comment (jschueller) [reply lib/src/Uncertainty/Distribution/TrapezoidalFactory.cxx]: I guess I could use a LinearFunction.

Comment (jschueller) [reply lib/src/Uncertainty/Distribution/TrapezoidalFactory.cxx]: done

Comment (sofianehaddad) [reply lib/src/Uncertainty/Distribution/TrapezoidalFactory.cxx]: ok

**Comment by regislebrun** [comment]
+1

**Pull request merged**